test/e2e/elasticache/replication_group/coverage_tests/rg_scaling.sh

```diff
@@ -0,0 +1,286 @@
+        - us-east-1b
+        replicaCount: 1
+      - nodeGroupID: "0020"
+        primaryAvailabilityZone: us-east-1b
+        replicaAvailabilityZones:
+        - us-east-1a
+        - us-east-1c
+        replicaCount: 2
+      - nodeGroupID: "0030"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+        - us-east-1b
+        replicaCount: 1
+EOF
+)
+  output_msg=$(echo "$rg_yaml" | kubectl apply -f - 2>&1)
+  exit_if_rg_config_application_failed $? "$rg_id"
+
+  # wait and assert new resource state
+  wait_and_assert_replication_group_available_status
+  k8s_assert_replication_group_shard_count "$rg_id" 3
+  k8s_assert_replication_group_total_node_count "$rg_id" 7
+}
+
+# basic scale out test for cluster mode enabled replication groups, # replicas/node group unchanged
+test_modify_rg_cme_scale_out_basic() {
+  debug_msg "executing ${FUNCNAME[0]}"
+
+  # generate and apply yaml for replication group creation
+  clear_rg_parameter_variables
+  rg_id="rg-cme-scale-out-basic"
+  num_node_groups="2"
+  replicas_per_node_group="1"
+  yaml_base="$(provide_replication_group_yaml)"
+  rg_yaml=$(cat <<EOF
+$yaml_base
+    nodeGroupConfiguration:
+      - nodeGroupID: "0010"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+        - us-east-1b
+      - nodeGroupID: "0020"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+        - us-east-1b
+EOF
+)
+  output_msg=$(echo "$rg_yaml" | kubectl apply -f - 2>&1)
+  exit_if_rg_config_application_failed $? "$rg_id"
+
+  # ensure resource successfully created and available, check resource is as expected
+  wait_and_assert_replication_group_available_status
+  k8s_assert_replication_group_shard_count "$rg_id" 2
+  k8s_assert_replication_group_replica_count "$rg_id" 1
+  k8s_assert_replication_group_total_node_count "$rg_id" 4
+
+   # update config and apply: scale out
+  num_node_groups="3"
+  yaml_base="$(provide_replication_group_yaml)"
+  rg_yaml=$(cat <<EOF
+$yaml_base
+    nodeGroupConfiguration:
+      - nodeGroupID: "0010"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+        - us-east-1b
+      - nodeGroupID: "0020"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+        - us-east-1b
+      - nodeGroupID: "0030"
+        primaryAvailabilityZone: us-east-1a
+        replicaAvailabilityZones:
+        - us-east-1b
+EOF
+)
+  output_msg=$(echo "$rg_yaml" | kubectl apply -f - 2>&1)
+  exit_if_rg_config_application_failed $? "$rg_id"
+
+  # wait and assert resource state
+  wait_and_assert_replication_group_available_status
+  k8s_assert_replication_group_shard_count "$rg_id" 3
+  k8s_assert_replication_group_replica_count "$rg_id" 1
+  k8s_assert_replication_group_total_node_count "$rg_id" 6
+}
+
+# run tests
+test_modify_rg_cmd_scale_out # currently failing, terminal condition frequently toggles (is this desired behavior?)
```

Comment:
maybe place this test in `test/e2e/elasticache/replication_group/update_fail.sh`?